plugins/rapid7_insightvm/komand_rapid7_insightvm/triggers/scan_completion/trigger.py

```diff
@@ -0,0 +1,267 @@
+            f"JOIN dim_vulnerability_reference AS dvr ON (fasvi.vulnerability_id = dvr.vulnerability_id) "
+            f"JOIN dim_solution AS dss ON (dv.nexpose_id = dss.nexpose_id) "
+            f"WHERE fasvi.scan_id = {scan_id} "
+        )
+
+
+class Cache:
+    @staticmethod
+    def get_cache_site_scans() -> dict:
+        try:
+            return json.loads(util.read_from_cache(ScanCompletion.CACHE_FILE_NAME))
+        except ValueError as error:
+            raise PluginException(cause="Failed to load cache file", assistance=f"Exception returned was {error}")
+
+
+class Util:
+    @staticmethod
+    def filter_results(params: dict, csv_row: dict, results: dict) -> Union[None, dict]:
+        """
+        Filter the outputted results based on the user inputs.
+
+        :param params: Input params
+        :param csv_row: Dict row of the csv results
+        :param results: New object to append results to
+
+        :return: New object containing only the necessary fields for the required output.
+        """
+
+        # Input retrieval
+        asset_group = params.get(Input.ASSET_GROUP, None)
+        cve = params.get(Input.CVE, None)
+        source = params.get(Input.SOURCE, None)
+        cvss_score = params.get(Input.CVSS_SCORE, None)
+        severity = params.get(Input.SEVERITY, None)
+
+        # We retrieve this separately because we use it as a unique identifier for
+        # the filtering process
+        asset_id = int(csv_row.get("asset_id", 0))
+
+        new_dict = {
+            "asset_id": asset_id,
+            "vulnerability_info": [
+                {
+                    "vulnerability_id": csv_row.get("vulnerability_id", ""),
+                    "nexpose_id": csv_row.get("nexpose_id", ""),
+                    "cvss_v3_score": csv_row.get("cvss_v3_score", 0),
+                    "severity": csv_row.get("severity", ""),
+                    "solution_id": Util.strip_msft_id(csv_row.get("solution_id", "")),
+                    "solution_summary": csv_row.get("summary", ""),
+                }
+            ],
+        }
+
+        # If an input and it is not found, return None in place of the row to filter
+        # out the result
+        if asset_group and asset_group not in csv_row.get("asset_group_id", ""):
+            return None
+        if cve and cve not in csv_row.get("nexpose_id", ""):
+            return None
+        if source and source not in csv_row.get("source", ""):
+            return None
+        if cvss_score and cvss_score < csv_row.get("cvss_v3_score", 0):
+            return None
+        if severity and severity not in csv_row.get("severity", ""):
+            return None
+        # Otherwise, return the newly filtered result.
+
+        existing_asset_id = results.get(asset_id, None)
+
+        if existing_asset_id:
+            existing_asset_id["vulnerability_info"] += new_dict["vulnerability_info"]
```

Comment:
would this be better as

` new_dict.get("vulnerability_info", [])`

in the event that its not set